OdooRpcWrapper/OdooAPI.cs

```diff
@@ -0,0 +1,130 @@
+﻿using CookComputing.XmlRpc;
+using OdooRpcWrapper.Exceptions;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+
+namespace OdooRpcWrapper
+{
+    public partial class OdooAPI : IOdooAPI
+    {
+        private const string STR_Id = "id";
+        private readonly OdooConnectionCredentials _credentials;
+        private readonly WebProxy _networkProxy;
+        private readonly RemoteCertificateValidationCallback _certificateValidationFunction;
+        private readonly List<Type> _validModels;
+        private IOdooObjectRpc _objectRpc;
+        private OdooLoginResult _loginResult = OdooLoginResult.Unknown;
+        
+        
+        public OdooAPI(OdooConnectionCredentials credentials, WebProxy networkProxy = null,
+            RemoteCertificateValidationCallback certificateValidationFunction = null)
+        {
+            _certificateValidationFunction = certificateValidationFunction;
+            _networkProxy = networkProxy;
+            _credentials = credentials;
+            _validModels = new List<Type>();
+        }
+
+        public OdooLoginResult Login()
+        {
+            IOdooCommonRpc loginRpc = XmlRpcProxyGen.Create<IOdooCommonRpc>();
+            loginRpc.Url = _credentials.CommonUrl;
+
+            if (_networkProxy != null)
+            {
+                loginRpc.Proxy = _networkProxy;
+            }
+
+            if (_certificateValidationFunction != null)
+            {
+                ServicePointManager.ServerCertificateValidationCallback = _certificateValidationFunction;
+            }
+            else
+            {
+                ServicePointManager.ServerCertificateValidationCallback = StandardValidationFunction;
+            }
+
+            // Log in and get user id
+            object result = null;
+            try
+            {
+                result = loginRpc.login(_credentials.DbName, _credentials.DbUser, _credentials.DbPassword);
+
+                if (result is bool)
+                {
+                    // Login failed!
+                    _credentials.UserId = -1;
+                    _loginResult = OdooLoginResult.InvalidCredentials;
+                }
+                else if (result is int)
```

Comment:
Don't use **else if** if it's possible . It's hard to read when the code is too long